packages/core/src/domain/conversion/upload-conversion-steps.ts

```diff
@@ -0,0 +1,220 @@
+import { Bundle, Resource } from "@medplum/fhirtypes";
+import { errorToString } from "@metriport/shared";
+import { SQSRecord } from "aws-lambda";
+import { S3Utils, executeWithRetriesS3 } from "../../external/aws/s3";
+import { capture } from "../../util";
+import { FHIR_APP_MIME_TYPE, XML_APP_MIME_TYPE } from "../../util/mime";
+import { buildDocumentNameForPartialConversions } from "./filename";
+
+export const defaultS3RetriesConfig = {
+  maxAttempts: 3,
+  initialDelay: 500,
+};
+
+export async function storePreProcessedConversionResult({
+  s3Utils,
+  conversionResult,
+  conversionResultBucketName,
+  conversionResultFilename,
+  message,
+  context,
+  lambdaParams,
+  log,
+}: {
+  s3Utils: S3Utils;
+  conversionResult: Bundle<Resource>;
+  conversionResultBucketName: string;
+  conversionResultFilename: string;
+  message: SQSRecord;
+  context: string;
+  lambdaParams: Record<string, string | undefined>;
+  log: typeof console.log;
+}) {
+  try {
+    await storeInS3WithRetries({
+      s3Utils,
+      payload: JSON.stringify(conversionResult),
+      bucketName: conversionResultBucketName,
+      fileName: conversionResultFilename,
+      contentType: FHIR_APP_MIME_TYPE,
+      log,
+    });
+  } catch (error) {
```

Comment:
Since we call `storeInS3WithRetries()` multiple times throughout this file, maybe we move error handling there and pass the msg/context and whether to call `capture.xxx()` or not?